src/yaml2candid/src/lib.rs

```diff
@@ -3,19 +3,36 @@
 #![deny(clippy::panic)]
 #![deny(clippy::unwrap_used)]
 use anyhow::{anyhow, bail, Context};
-use candid::types::value::{IDLField, IDLValue, VariantValue};
+use candid::{
+    types::value::{IDLField, IDLValue, VariantValue},
+    Principal,
+};
 use candid_parser::{
     types::{Dec, IDLType},
     IDLProg,
 };
+use num_bigint::{BigInt, BigUint};
 use serde_yaml::Value as YamlValue;
 use std::path::Path;
+#[cfg(test)]
+mod test;
 
 /// Converts YAML to Candid using a given did file.
 pub struct Yaml2Candid {
     /// The types that the converter supports, defined in an IDLProg interface definition.
     pub prog: IDLProg,
 }
+impl Default for Yaml2Candid {
```

Comment:
Simple default constructor needed for tests.  Included here as it may have more uses.